package-spotlight-class-metadata.md

```diff
@@ -0,0 +1,210 @@
+
+```php
+Alias::classFor('product-category'); // "App\Entity\ProductCategory"
+```
+
+After querying our database for activity, we can use this to get the class name
+from the stored alias:
+
+```php
+class Activity
+{
+    // ...
+
+    public function getEntityClass(): ?string
+    {
+        return Alias::classFor($this->entityAlias);
+    }
+
+    // ...
+}
+```
+
+***NOTE
+`Alias::classFor()` will return `null` if no class is found for the given alias.
+***
+
+## Alternative Alias Configuration
+
+What about classes you don't control, like vendor classes? No problem! You can configure
+these in your app's `composer.json` file:
+
+```json
+{
+    "extra": {
+        "class-metadata": {
+            "map": {
+                "Galactic\\Fleet\\Hyperdrive": "hyperdrive"
+            }
+        }
+    }
+}
+```
+
+## Additional Metadata
+
+Say we don't want *all* entities to be auditable - they need to be *marked* as such.
+We have a couple options:
+
+1. Use a marker interface (an interface without methods), e.g. `AuditableInterface`, and
+   have auditable entities implement it.
+2. Create a custom class attribute, e.g. `#[Auditable]`, and add it to auditable entities.
+
+These both work, but this package provides a third option: the `#[Metadata]` attribute:
+
+```php
+use Zenstruck\Metadata;
+
+#[Metadata('auditable', true)]
+class ProductCategory
+{
+    // ...
+}
+```
+
+***IMPORTANT
+Metadata is also picked up by the Composer plugin during the autoload dump process. Be sure to run:
+
+```terminal
+composer dump-autoload
+```
+
+whenever you add or change metadata.
+***
+
+Now, before logging activity, we first check if the entity is auditable:
+
+```php
+use Zenstruck\Metadata;
+
+function onEntityChange(object $entity, int $entityId, string $action): void
+{
+    if (!Metadata::get($entity, 'auditable')) {
+        return; // not auditable
+    }
+
+    $this->logActivity($entity, $entityId, $action);
+}
+```
+
+***NOTE
+`Metadata::get()` will return `null` if the given key is not found for the class or object.
+***
+
+Check out the [Metadata Lookup documentation](https://github.com/zenstruck/class-metadata/blob/1.x/README.md#metadata-lookup)
+for more advanced usages.
+
+## Listing all Aliases and Metadata
+
+To keep track of all your configured aliases and metadata, the package's Composer plugin
+provides a command to list them all:
+
+```terminal
+composer list-class-metadata
+```
+
+The output looks something like this:
+
+```
+ ---------------------------- ------------------ --------------------
+  Class                        Alias              Metadata
+ ---------------------------- ------------------ --------------------
+  App\Entity\ProductCategory   product-category   {"auditable":true}
+ ---------------------------- ------------------ --------------------
+```
+
+---
+
+Whether you need simple aliases for storage, or metadata to drive business logic, this package
+keeps things fast, organized, and clean.
+
```

Comment:
Maybe say a few words about where it can be useful? So far, we mostly explain how to configure it, but sharing some use-case ideas would be nice IMO.